### PR TITLE
feat: [PLG-1537]: Don't allow monthly subscriptions to purchase premium sup…

### PR DESCRIPTION
**Pull request by thomas-grasman**
…port.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38601)

### CODE REVIEW

**Comment by thomas-grasman** [comment]
trigger messagecheck

File: 930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/SubscriptionServiceImpl.java [hunk 1]
```diff
@@ -196,6 +197,9 @@ public SubscriptionDetailDTO createFfSubscription(String accountIdentifier, FfSu
     subscriptionItems.add(ItemParams.builder().priceId(mauPriceId.getId()).quantity(1L).build());
 
     if (subscriptionDTO.isPremiumSupport()) {
+      if (subscriptionDTO.getPaymentFreq().trim().equalsIgnoreCase("monthly")) {
```
Comment (pranil-harness) [inline 930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/SubscriptionServiceImpl.java:200]: a lot of magic strings in the code here - MAU, monthly, DEVELOPERS, etc. Please consider moving them to a constants file

**Review by pranil-harness — COMMENTED** [review]
> Reviewed 1 of 1 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38601)* status: 2 file(s) to review. 0 of 2 approvals obtained.

**Review by pranil-harness — APPROVED** [review]

**Review by pranil-harness — APPROVED** [review]

**Review by pranil-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38601#-:-NE4fvzN4pOhf-uvgqX6:bnfp4nl)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by pranil-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38601#-:-NE4g2pN7juJQ_G3G5JD:bnfp4nl)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by pranil-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38601#-:-NE4g5op56JUB0vc4ufY:bnfp4nl)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Pull request merged**